oscarapi/views/basket.py

```diff
@@ -92,13 +98,20 @@ def post(self, request, format=None):
             quantity = p_ser.validated_data['quantity']
             options = p_ser.validated_data.get('options', [])
 
-            basket_valid, message = self.validate(basket, product, quantity, options)
+            basket_valid, message = self.validate(
+                basket, product, quantity, options)
             if not basket_valid:
                 return Response(
                     {'reason': message},
                     status=status.HTTP_406_NOT_ACCEPTABLE)
 
-            basket.add_product(product, quantity=quantity, options=options)
+            line, created = basket.add_product(
```

Comment:
not really sure about this. Doesn't this encourage people to update basket quantity with negative values?